dvc/repo/metrics/show.py

```diff
@@ -23,16 +23,23 @@ def _collect_metrics(repo, targets, recursive):
     if not targets:
         return list(metrics)
 
+    result = []
     target_infos = [PathInfo(os.path.abspath(target)) for target in targets]
+    if recursive:
```

Comment:
Will recursive work with unofficial metrics? I don't see a test for it and the code seems like it won't work.

Reply:
I am not sure how that would work, I mean, in that case we would need to try to show all files for particular directory, since they are not marked as metrics. So the question here is whether we want to support that.

Reply:
@pared That seems like a natural approach. You have a test with official metrics dir in this PR, what changes if they are not declared as official metrics?

Reply:
On the other hand, it kinda makes sense to have special dir for metrics and check all of them out. Ill try to modify the change.

Reply:
Also note that we try to parse metric files, so if something is not a metric we'll just ignore it and won't show it.